### PR TITLE
Moved iree/modules/check/iree-check-module-main.cc to iree/tools/iree-check-module-main.cc: Issue 1986

### PR DESCRIPTION
**Pull request by inho9606**
Updated: build_tools/bazel/iree_check_test.bzl, iree/modules/check/test/BUILD
Moved iree-check-module-main.cc from iree/modules/check to iree/tools
Modified BUILD files in iree/tools/ and iree/modules/check/
Modified CMakeList.txt n iree/tools and iree/modules/check/
Modified iree/modules/check/test/CMakeLists.txt
Modified iree/modules/check/test/BUILD
Modified build_tools/cmake/iree_check_test.cmake
Modified build_tools/bazel/iree_check_test.bzl
I hope it works please.
Thanks

### CODE REVIEW

File: SUBMODULE_VERSIONS [hunk 1]
```diff
@@ -4,8 +4,13 @@ daff5fead3fbe22c6fc58310ca3f49caf117f185 third_party/benchmark
 4fb0ff7069bd88ee85902f4d0bb62794e5f6d021 third_party/flatcc
 b1fbd33c06cdb0024c67733c6fdec2009d17b384 third_party/googletest
 76a21602d84dd4b1a119523954df0c1686d1a4d6 third_party/llvm-bazel
+<<<<<<< HEAD
 d553243fe4b5e1992c07aff7b54b16160a4d5e97 third_party/llvm-project
 a438d3b2e9e2e4699b94787b7fb6e8b45d4e92c6 third_party/mlir-emitc
+=======
+6883042528d0338b776bb2316e58999650e94892 third_party/llvm-project
+55801f03f9cc69abfcf8b508a873f702c11b3b5f third_party/mlir-emitc
+>>>>>>> 4110f3daae21078bf885a618e6789033684a675a
```
Comment (hanhanW) [inline SUBMODULE_VERSIONS:13]: This looks like a conflict, please fix it.

File: integrations/tensorflow/compiler/BUILD [hunk 2]
```diff
@@ -114,7 +114,7 @@ cc_binary(
         ":open_source_build": [],
         "//conditions:default": [
             # TODO: Enable this for all builds once dependency bug is fixed.
-            #"@org_tensorflow//tensorflow/compiler/mlir/tensorflow:tf_dialect_passes",  # Note: always_links in random foo that makes constant folding work
+            "@org_tensorflow//tensorflow/compiler/mlir/tensorflow:tf_dialect_passes",  # Note: always_links in random foo that makes constant folding work
```
Comment (hanhanW) [inline integrations/tensorflow/compiler/BUILD:117]: I think we don't need this change, let's revert it.

**Review by hanhanW — APPROVED** [review]
nice!

File: SUBMODULE_VERSIONS (hunk 1 above)
Comment (hanhanW) [reply SUBMODULE_VERSIONS:13]: I think we don't need to update this file, let's just revert it.

**Comment by hanhanW** [comment]
Hey @inho9606, buildifier is for formatting BUILD. The failure happened because files were not matched to formatting results. I have a patch to fix BUILD files, please take a look. https://gist.github.com/hanhanW/ace515f457bcb456d9682dad9a9996f6

You can use command `git apply fixed_build.patch` to fix it.

**Comment by inho9606** [comment]
> Hey @inho9606, buildifier is for formatting BUILD. The failure happened because files were not matched to formatting results. I have a patch to fix BUILD files, please take a look. https://gist.github.com/hanhanW/ace515f457bcb456d9682dad9a9996f6
> 
> You can use command `git apply fixed_build.patch` to fix it.

**Comment by inho9606** [comment]
Hi @hanhanW 
Thanks for the patch file! I successfully fixed the fail.

**Comment by inho9606** [comment]
Don't I need to combine 7 commits into 1 commit? If so, how can I do it?

**Pull request merged**